src/trading/ClaimProceeds.sol

```diff
@@ -0,0 +1,56 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+import 'ROOT/Mutex.sol';
+import 'ROOT/reporting/Interfaces.sol';
+import 'ROOT/extensions/MarketFeeCalculator.sol';
+import 'ROOT/legacy_reputation/SafeMath.sol';
+
+
+/**
+ * @title ClaimProceeds
+ * @dev This allows users to claim their money from a market by exchanging their shares 
+ * AUDIT: Ensure that a malicious market can't subversively cause share tokens to be paid out incorrectly.
+ */
+contract ClaimProceeds is Controlled {
+    using SafeMath for uint256;
+
+    function publicClaimProceeds(IMarket market) onlyInGoodTimes public returns(bool) {
```

Comment:
Style: All local variables and parameters should be prefixed with `_`.  This is to avoid name collisions with storage variables as well as make it clear to the viewer the scope of the variable.